metrics-api/src/main/java/software/amazon/swage/metrics/NullContext.java

```diff
@@ -0,0 +1,48 @@
+package software.amazon.swage.metrics;
+
+import java.time.Instant;
+
+import software.amazon.swage.collection.TypedMap;
+
+/**
+ * A MetricContext implementation that acts as a sink for recorded events.
+ */
+public class NullContext implements MetricContext {
+
+    private static final NullContext EMPTY = new NullContext(TypedMap.empty());
+
+    /**
+     * The default NullContext with an empty set of attributes.
+     *
+     * @return the empty NullContext
+     */
+    public static NullContext empty() {
+        return EMPTY;
+    }
+
+    private final TypedMap attributes;
+
+    /**
+     * @param attributes the attributes of the measurement context
+     */
+    public NullContext(TypedMap attributes) {
+        this.attributes = attributes;
```

Comment:
Is this actually necessary? Seems like simply letting this go out of scope would be more reasonable.